pkg/maglev/maglev.go

```diff
@@ -74,33 +75,60 @@ func getPermutation(backends []string, m uint64) []uint64 {
 	return perm
 }
 
+// BackendPoint is a backend point with weight
+type BackendPoint struct {
+	ID     uint16
+	Weight uint32
+}
+
 // GetLookupTable returns the Maglev lookup table of the size "m" for the given
 // backends. The lookup table contains the indices of the given backends.
-func GetLookupTable(backends []string, m uint64) []int {
-	if len(backends) == 0 {
-		return nil
+// maglevBackendIDsBuffer. The slice contains backend IDs.
+func GetLookupTable(backends map[string]*BackendPoint, m uint64, maglevBackendIDsBuffer []uint16) {
```

Comment:
Didn't catch this the first time around, but I noticed that we no longer return from this function anymore. Could you update the godoc accordingly?

Reply:
OK. I have updated the descriptions.